src/librustc_mir/interpret/eval_context.rs

```diff
@@ -869,6 +869,9 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         // Our result will later be validated anyway, and there seems no good reason
         // to have to fail early here.  This is also more consistent with
         // `Memory::get_static_alloc` which has to use `const_eval_raw` to avoid cycles.
+        // FIXME: We can hit delay_span_bug if this is an invalid const, interning finds
+        // that problem, but we never run validation to show an error. Can we ensure
+        // this does not happen?
```

Comment:
I still need to try and actually case the ICE here -- but my worry is that the `delay_span_bug` in interning all assume that validation will be run after interning is done, and this code here violates that assumption.

Reply:
Turns out we can get that to ICE even without Miri:
```rust
#![feature(const_transmute)]
#![warn(const_err)]

use std::mem;

const X: &i32 = unsafe {
    let x = 0;
    mem::transmute(&x)
};
```

Reply:
Do you want to add a test for this in this PR or fix it in this PR?

Reply:
The PR [does contain a test for the ICE outside Miri](https://github.com/rust-lang/rust/pull/71665/files#diff-c27257d12985eefbccba2a933eca48f4) -- mostly accidentally, when I added `allow(const_err)` I did not realize I was testing an ICE. ;)

For whether this can ICE with Miri, I haven't done another attempt at triggering it.